Code/GraphMol/FindRings.cpp

```diff
@@ -856,21 +865,29 @@ bool findRingConnectingAtoms(const ROMol &tMol, const Bond *bond,
 
 namespace RDKit {
 namespace MolOps {
+int findSSSR(const ROMol &mol, VECT_INT_VECT &res, bool includeDativeBonds) {
+  return findSSSR(mol, *mol.getRingInfo(), &res, includeDativeBonds);
```

Comment:
This highlights what I see as one of the big problems with this change:
If I construct a molecule without santiziation and then call `findSSSR()`, it's going to call the symmetric ring finding code (since getRingInfo() will be called on an uninitialized structure), then throw that information away and call the SSSR code. That's really not what we want.

The other big problem is that all existing RDKit code which uses this currently recommended pattern (or the python equivalent):
```
if(!mol.getRingInfo() || !mol.getRingInfo()->isInitialized()){
  // maybe find the rings the way the code expects, maybe do something else
}
```
is going to always end up with whatever the default ring finding operation is and will never enter the loop. That's a significant behavior change.

Reply:
I'm not sure we don't have this issue anyway, if the ring is already initialized you have no idea how it was initialized (except with the new atropisomer code).

As you can tell, I'm not a big fan of the current behavior because of the const issues.  It seems to me that we have a few solutions.

```
mol.getRingInfo(RingInfo::SSSR)
mol.getRingInfo(RingInfo::SYMSSSR)
mol.getRingInfo(RingInfo::FASTSSSR)

mol.hasRingInfo(RingInfo::SSSR)
mol.hasRingInfo(RingInfo::SYMSSSR)
mol.hasRingInfo(RingInfo::FASTSSSR)
```

And we can modify RingInfo to store these separately or just store them in the molecule if they are created, it's just three potential pointers.

Or ignore it and keep the things the same.

However, I do think being able to make the rings without storing them on the molecule is pretty useful, i.e. if you want the SymSSSR rings but the molecule already has the normal ones.

Reply:
I.e. adding the ring finding type to RingInfo is a good idea.